lisa/runner.py

```diff
@@ -247,6 +256,27 @@ def _callback_completed(self, results: List[TestResult]) -> None:
         finally:
             self._results_lock.release()
 
+    def _submit_tasks(
+        self,
+        remaining_runners: List[BaseRunner],
+        task_manager: TaskManager[List[TestResult]],
+    ) -> None:
+        for runner in remaining_runners:
```

Comment:
The runner loop can be left in the main loop method, and the method called `_submit_runner_tasks`. The logic maybe more clear, because the remaining_runners are maintained in the main loop method.